pkg/cmd/roachtest/tests/db_console_endpoints.go

```diff
@@ -0,0 +1,419 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package tests
+
+import (
+	"context"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/rand"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/roachtestutil"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/roachtestutil/mixedversion"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/spec"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/install"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/logger"
+	"github.com/cockroachdb/cockroach/pkg/server/serverpb"
+	"github.com/cockroachdb/cockroach/pkg/util/retry"
+	"github.com/cockroachdb/errors"
+)
+
+//go:embed db-console/admin_endpoints.json
+var adminEndpointsJSON string
+
+//go:embed db-console/api_v2_endpoints.json
+var apiV2EndpointsJSON string
+
+//go:embed db-console/status_endpoints.json
+var statusEndpointsJSON string
+
+type HTTPMethod string
+
+const (
+	GET  HTTPMethod = "GET"
+	POST HTTPMethod = "POST"
+)
+
+// endpoint represents a DB console endpoint to test.
+type endpoint struct {
+	url            string
+	method         HTTPMethod
+	verifyResponse *func(resp *http.Response) error
+}
+
+// endpointJSON represents the common structure for all endpoint JSON files
+type endpointJSON struct {
+	Endpoints []struct {
+		URL  string `json:"url"`
```

Comment:
Since verifyResponse doesn't seem to be used right now, what if we just had the endpoint definition above match this json parseable version:

```
 // Endpoint represents a DB console endpoint to test.
type Endpoint struct {
	URL  string `json:"url"`
	Verb string `json:"verb"`
	Skip bool   `json:"skip,omitempty"`
}

// EndpointJSON represents the common structure for all endpoint JSON files
type EndpointJSON struct {
	Endpoints []Endpoint `json:"endpoints"`
}
```

Reply:
done.